js/reference.js

```diff
@@ -1020,6 +1020,21 @@ var ERMrest = (function(module) {
                     }
                     var page = new Page(ownReference, etag, response.data, hasPrevious, hasNext);
 
+                    // we are paging based on @before (user navigated backwards in the set of data)
+                    // AND there is less data than limit implies (beginning of set) OR we got the right set of data (tuples.length == pageLimit) but there's no previous set (beginning of set)
+                    if ( (ownReference.location.pagingObject && ownReference.location.pagingObject.before) && (page.tuples.length < limit || !page.hasPrevious) ) {
+                        // should I clone the reference then set the paging object to null?
+                        ownReference.location.pagingObject = null;
+                        ownReference.read().then(function rereadReference(page) {
+                            defer.resolve(page);
+                        }, function error(response) {
+                            var error = module._responseToError(response);
+                            return defer.reject(error);
+                        }).catch(function (error) {
```

Comment:
Now that we are working on error handling, I don't think this catch would be needed. The [outer catch](https://github.com/informatics-isi-edu/ermrestjs/blob/96a5d11da44c7cc1acc08b46feff015c4f45522e/js/reference.js#L1043) is doing the same.